### PR TITLE
feat: [IOBP-348] Addition of bottomSheet in idPay discount initiative CTA

### PR DESCRIPTION
**Pull request by forrest57**
## Short description

DEPENDS ON https://github.com/pagopa/io-app/pull/5212

addition of bottomSheet in idPay discount type initiatives in sight of the future inclusion of barcode generation

## List of changes proposed in this pull request
- addition of hook to handle the bottomSheet
- removal of qrcode icon from CTA as of design
- addition of required I18n

## How to test
navigate to an idpay discount initiative, press "authorize payment" and make sure that:
- a bottomSheet appears
- "scan QR" navigates to the current qrcode scan screen
- "create barcode" closes the bottomSheet
- a11y keys are correctly read.

### CODE REVIEW

File: ts/features/idpay/details/utils/hooks.tsx [hunk 1]
```diff
@@ -82,3 +91,47 @@ export const useInitiativeTimelineFetcher = (
     lastUpdate
   } as const;
 };
+
+export const useIdpayDiscountDetailsBottomSheet = () => {
```
Comment (mastro993) [inline ts/features/idpay/details/utils/hooks.tsx:95]: If this bottom-sheet is only used inside `InitiativeDetailsScreen.tsx`, what is the point of making it an utility function accessible from anywhere? Do you think this is a good design?

Comment (forrest57) [reply ts/features/idpay/details/utils/hooks.tsx]: honestly, I just don't like how long that screen was getting so I thought moving some parts outside would greatly improve readability for the main screen.
What do you think? yes adding another export to the codebase could be not so graceful, but so is adding another ~60 lines to an already huge file

File: ts/features/idpay/details/utils/hooks.tsx [hunk 2]
```diff
@@ -82,3 +92,53 @@ export const useInitiativeTimelineFetcher = (
     lastUpdate
   } as const;
 };
+
+export const useIdpayDiscountDetailsBottomSheet = (initiativeId: string) => {
```
Comment (mastro993) [inline ts/features/idpay/details/utils/hooks.tsx:96]:
```suggestion
export const useIdPayDiscountDetailsBottomSheet = (initiativeId: string) => {
```

Another thing: imo this `hooks.ts` file is growing pretty fast. What do you think about a `hooks` folder where to place all hooks inside?
In this case we would have:
- `ts/features/idpay/details/hooks/useIdPayDiscountDetailsBottomSheet.tsx`

**Review by mastro993 — CHANGES_REQUESTED** [review]

File: ts/features/idpay/details/utils/hooks.tsx (hunk 2 above)
Comment (forrest57) [reply ts/features/idpay/details/utils/hooks.tsx:96]: sure, I don't see any reason why not 
That said, can I mark your previous comment as resolved? to me it seems like this one resolves the last one, LMK👍

Comment (forrest57) [reply ts/features/idpay/details/utils/hooks.tsx]: addressed in [064d125](https://github.com/pagopa/io-app/pull/5146/commits/064d12534a39bcb8dcf1130b00d8002d23f01c45)

**Review by mastro993 — APPROVED** [review]
LGTM

**Pull request merged**